### PR TITLE
Make create operation use GTask API to use native Vala async support

### PR DESCRIPTION
**Pull request by tintou**
This will also ease the port of the remaining parts to Vala

### CODE REVIEW

File: src/View/AbstractDirectoryView.vala [hunk 1]
```diff
@@ -939,19 +939,36 @@ namespace FM {
...
             /* Block the async directory file monitor to avoid generating unwanted "add-file" events */
             slot.directory.block_monitor ();
-            Marlin.FileOperations.new_file (this as Gtk.Widget,
-                                            null,
-                                            parent_uri,
-                                            null,
-                                            null,
-                                            0,
-                                            create_file_done);
+            Marlin.FileOperations.new_file (
```
Comment (jeremypw) [inline src/View/AbstractDirectoryView.vala]: `.begin` missing?

File: src/View/AbstractDirectoryView.vala [hunk 2]
```diff
@@ -2411,12 +2428,21 @@ namespace FM {
             /* Block the async directory file monitor to avoid generating unwanted "add-file" events */
             slot.directory.block_monitor ();
             var new_name = (_("Untitled %s")).printf (template.get_basename ());
-            Marlin.FileOperations.new_file_from_template (this,
-                                                          null,
-                                                          slot.location,
-                                                          new_name,
-                                                          template,
-                                                          create_file_done);
+            Marlin.FileOperations.new_file_from_template (
```
Comment (jeremypw) [inline src/View/AbstractDirectoryView.vala]: `.begin` missing?

**Comment by jeremypw** [comment]
Looks good.  I'll give it some testing.

File: src/View/AbstractDirectoryView.vala (hunk 1 above)
Comment (tintou) [reply src/View/AbstractDirectoryView.vala]: oops, indeed, it was written with the ton of warnings :)

**Pull request merged**